src/components/services/OctoPi.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <Generic :item="item">
+    <template #content>
+      <p class="title is-4">{{ item.name }}</p>
+      <p class="subtitle is-6">
+        <template v-if="item.subtitle && printerstatus == null">
+          {{ item.subtitle }}
+        </template>
+        <template v-else-if="api" && printerstatus !="null">
+          Status: {{ printerstatus }}
+        </template>
+        <template v-if="printerstatus == 'Printing'"
+          >|| Progress: {{ printerprogress }}% <br />
+          Elapsed: {{ printtime }} || Left: {{ printtimeleft }}
+        </template>
+      </p>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import service from "@/mixins/service.js";
+import Generic from "./Generic.vue";
+
+export default {
+  name: "OctoPi",
+  mixins: [service],
+  props: {
+    item: Object,
+  },
+  components: {
+    Generic,
+  },
+  data: () => ({
+    api: {
+      code: true,
+      state: "",
+      job: {
+        file: {
+          display: "",
+        },
+      },
+      progress: {
+        state: "",
+        completion: 0,
+        printTimeLeft: 0,
+        printTime: 0,
+      },
+    },
+  }),
+  computed: {
+    printerstatus: function () {
+      if (this.api && this.api.code != false) {
+        return this.api.state;
+      } else if (this.api.code != false) {
+        return "Offline";
+      }
+      return null;
+    },
+    printerprogress: function () {
+      if (this.api) {
+        return this.api.progress.completion.toFixed();
+      }
+      return 0;
+    },
+    printtime: function () {
+      if (this.api) {
+        var timestamp = this.api.progress.printTime;
+        var hours = Math.floor(timestamp / 60 / 60);
+        var minutes = Math.floor(timestamp / 60) - hours * 60;
+        var seconds = timestamp % 60;
+        if (hours < 10) {
+          hours = "0" + hours;
+        }
+        if (minutes < 10) {
+          minutes = "0" + minutes;
+        }
+        if (seconds < 10) {
+          seconds = "0" + seconds;
+        }
+        var formatted = hours + ":" + minutes;
+        return formatted;
+      }
+      return 0;

```

Comment:
Invert the if block here in order to reduce nesting. That way the rest of the code in the function can have one less indent. See https://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting.

Additionally, please use `let` and `const` instead of `var`. The `var` keyword comes with a different scoping mechanism and can cause unintentional value bleed into other parts of the code. Here you would use `const` for `timestamp` and `formatted`, and `let` for the others (you could also skip declaring a `formatted` variable entirely and just return `${hours}:${minutes}`).

```
if (!this.api) {
  return 0;
}

// Rest of code here
const timestamp = ...
```

Here and below for printtimeleft